en/searching-multi-valued-fields.md

```diff
@@ -0,0 +1,603 @@
+
+<pre>
+query=[AND sunshot photos featuring dogs]
+</pre>
+Using tracing is very useful when debugging why documents match or does not match. 
+
+Since our sample document does not contain the term *featuring* or *photos*, the query fails to retrieve the example document. 
+We can relax the query matching to instead of requiring that **all** terms match, to use **any**. See
+[model.type](reference/query-api-reference.html#model.type) query api reference for supported query types.
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=sunset photos featuring dogs' 'type=any'
+</pre>
+
+Changing the type to `any`, recalls our sample document as we no longer require that all query terms must match.
+With `type` it also possible to require that indivudual query terms match by using `+`:
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=+sunset photos featuring +dogs' 'type=any'
+</pre>
+
+In this example `sunset` and `dogs` must be matched. Note that we have disabled stemming so querying
+for `dogs` won't recall documents with `dog`. This is one of the reasons we disabled stemming, to demonstrate
+that stemming has impact on recall. Requiring `dog` will cause the query to not recall our sample document.
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 0'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=+sunset photos featuring +dog' 'type=any'
+</pre>
+
+Now, let us explore how Vespa matches the multi-valued tags field of 
+type [weightedset](reference/schema-reference.html#weightedset). 
+Notice that we change back to default `type=all`. 
+In this example we also use the [default-index](reference/query-api-reference.html#model.defaultIndex) 
+query parameter to limit matching to the `tags` field.
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=clear sky' 'type=all' 'default-index=tags'
+</pre>
+
+The query matches the document which is no surprise since a tag contains the exact content `clear sky`.
+Let us search for just `clear` instead:
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=clear' 'type=all' 'default-index=tags'
+</pre>
+
+Also matches the document, this demonstrates that matching is partial, it does not require
+to match the set element exactly. `clear` matches `clear sky` and `sky` will match `clear sky`.  
+
+But what about `black sky`:
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=black sky' 'type=all' 'default-index=tags'
+</pre>
+
+Also matches our document. This is an example of cross-element matching. With weightedset
```

Comment:
consider linking to https://docs.vespa.ai/en/reference/query-language-reference.html#sameelement here, with a short note?

Reply:
Problem is that sameElement does not offer any ranking features :/

Reply:
right. I was thinking of just the matching part, and this is a frequent support question. I'll look at the doc again and see if there is a better place to just mention this

Reply:
sameElement is great, but only supported for the complex array of struct and map, which per today, is only supported for attribute fields, and has no rank feature support.